src/System.Runtime.InteropServices/src/System/Runtime/InteropServices/RuntimeEnvironment.cs

```diff
@@ -23,7 +23,10 @@ public static bool FromGlobalAccessCache(System.Reflection.Assembly a)
         }
         public static string GetRuntimeDirectory()
         {
-            return Path.GetDirectoryName(typeof(object).Assembly.Location) + Path.DirectorySeparatorChar;
```

Comment:
Yes, this is looking for machine.config. I inject an in memory one if one can't be found here. On NetFX this would be `C:\Windows\Microsoft.NET\Framework\v4.0.30319\` and we end up sticking `Config\machine.config` on the end.

In CoreFX this is wherever System.Private.Corelib is. I think the equivalent should probably be where the actual exe is for AOT? The idea is that you _can_ drop a custom machine.config in if you need to without resorting to changing APIs and manually passing in paths for your config files.